### PR TITLE
use testthat3e (main)

### PR DESCRIPTION
**Pull request by dpprdan**
This makes the necessary changes to use [testthat 3e](https://testthat.r-lib.org/articles/third-edition.html). 

- [x] Add NEWS item

### CODE REVIEW

File: tests/testthat/test-deprecated.R [hunk 1]
```diff
@@ -1,105 +1,106 @@
...
+  expect_type(results[["total_results"]], "integer")
+  expect_s3_class(results[["time_stamp"]], "POSIXct")
   expect_equal(length(results), 4)
   expect_equal(sum(grepl(
     "annotation",
     names(results[["results"]])
   )), 0)
   expect_true(dplyr::between(nrow(results[["results"]]), 1, 2))
 
-  lifecycle::expect_deprecated(
```
Comment (maelle) [inline tests/testthat/test-deprecated.R:47]: is there now any expectation around the deprecation in opencage tests?

**Review by maelle — APPROVED** [review]
Thank you, great work :rocket:

File: tests/testthat/test-deprecated.R (hunk 1 above)
Comment (dpprdan) [reply tests/testthat/test-deprecated.R:47]: Yes, there are still `lifecycle::expect_deprecated` calls in lines 4 and 13. I hid the explanation in this commit message [`1a0bab0` (#141)](https://github.com/ropensci/opencage/pull/141/commits/1a0bab0f7fa30584aa51ffb800ad240c07841036). 😄 

(Hope I understood your question correctly.)

Comment (maelle) [reply tests/testthat/test-deprecated.R:47]: aaah ok, yes, thanks!

**Pull request merged**